src/libs/antares/array/matrix.hxx

```diff
@@ -1552,6 +1552,18 @@ inline void Matrix<T, ReadWriteT>::copyFrom(const Matrix<U, V>* rhs)
         copyFrom(*rhs);
 }
 
+template<class T, class ReadWriteT>
+template<class U, class V>
+void Matrix<T, ReadWriteT>::swap(Matrix<U, V>& rhs)
+{
+  static_assert(Yuni::Static::Type::StrictlyEqual<T, U>::Yes);
+  using std::swap;
+  swap(this->width, rhs.width);
```

Comment:
If we swap matrices, their width and height must be equal, right ?
If not, we have a problem, don't you think ?

Reply:
`swap` should always succeed. Adding an assertion could potentially make it fail.

swapping matrices of different size makes sense. It is up to the user to check for size consistency.